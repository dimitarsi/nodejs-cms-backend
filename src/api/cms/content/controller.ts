import contentRepo from "@repo/content"
import { createCrudController } from "~/core/api/crud/controller"
import { getConfig } from "./actions/getConfig"
import { search } from "./actions/search"
import { crudRoutes } from "~/core/api/crud/router"

export const ContentController = createCrudController(contentRepo, {
  getConfig,
  search,
})

const routes = crudRoutes(`/content/`, ContentController)

routes.get("/search", ContentController("search"))
routes.get(":id/config", ContentController("getConfig"))

export default routes
