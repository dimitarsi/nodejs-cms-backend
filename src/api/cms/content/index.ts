import contentTypesRepo from '@repo/contentTypes';
import contentRepo from '@repo/content'
import defaultController from "~/core/api/controller"
import express, { Request, application, query } from "express"
import ajv from "~/schema/core"
import { type JSONSchemaType } from "ajv"
import { Validator } from '~/core/api/request';
import { ObjectId } from 'mongodb';

const app = express()

const validate = ajv.compile<
  JSONSchemaType<{
    configId: string
    name: string
    slug: string
    data: Record<string, any>
  }>
>({
  $async: true,
  type: "object",
  properties: {
    configId: {
      type: ["string", "null"],
      found: { in: contentTypesRepo.collectionName(), optional: true },
    },
    name: {
      type: "string",
    },
    slug: {
      type: "string",
      found: {
        in: "stories:slug",
        notIn: true,
      },
    },
    type: {
      enum: ["folder", "document"],
    },
    folder: {
      type: "string",
    },
    data: {
      type: "object",
      properties: {},
    },
  },
})

const validateRequestBody: Validator = (req: express.Request) => {
  return undefined

  // return validate(req.body).then(_result => {
  //   return undefined
  // }).catch((err) => {
  //   return err;
  // })
}

app.get('/search', async (req, res) => {
  const page = parseInt(req.query["page"]?.toString() || "1")
  const folderName = req.query["folder"] || '/'

  let folderQuery: Record<string, any> = {};
  
  if(typeof folderName === 'string') {
    folderQuery = {folderLocation: folderName}
  // @ts-ignore
  } else if(typeof folderName === 'object' && typeof folderName?.['like'] === 'string') {
    folderQuery = {
      // @ts-ignore
      folderLocation: new RegExp(`${folderName?.["like"]?.toString()}`),
    }
  // @ts-ignore
  } else if(typeof folderName === 'object' && typeof folderName?.['startsWith'] === 'string') {
    folderQuery = {
      folderLocation: new RegExp(
        // @ts-ignore
        `^${folderName?.["startsWith"]?.toString()}.?`
      ),
    }
  }

  res.json(
    await contentRepo.getAll(isNaN(page) ? 1 : Math.max(1, page), {
      pageSize: 20,
      filter: folderQuery,
    })
  )
});


app.get('/:id/config', async (req, res) => {
  const id = req.params["id"]

  const content = await contentRepo.getById(id)

  if (content == null) {
    res.status(404).json({
      message: "Could not find item with idOrSlug: " +id  
    })
    return;
  }

  const config = await contentTypesRepo.getById(content.configId)

  res.json(config)
})

defaultController(app, contentRepo, {
  create: validateRequestBody,
  // update: validateRequestBody,
})

export default app
