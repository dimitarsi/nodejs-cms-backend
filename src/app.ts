import "./config"
import cmsAPI from "~/api/cms"
import "~/middleware/auth"
import "~/middleware/devEnv"
import fastify, { FastifyPluginCallback } from "fastify"
import dbLayer from "~/repo"
import publicApiAuth from "~/api/public"
import fastifyPlugin from "fastify-plugin"
// import attachments from "@api/cms/files"
// import media from "@api/public/media"
import cmsApiSchema from "~/schema/cms-api"
// import mediaPlugin from "@api/media"

const app = fastify({
  logger: true,
})

// app.use(bodyParser.urlencoded())
// app.use(bodyParser.json())
// app.use(cors())

// const router = Router()
// publicAPI(router)

// app.router.use("auth:isAdmin")
// {
// app.register(dbConnector)

app.register(fastifyPlugin(cmsApiSchema))

app.register(fastifyPlugin(dbLayer), {
  dbName: process.env.DB_NAME || "plenty_cms",
})

app.get("/health", (_r, reply) => {
  reply.code(200).send({})
})

app.register(cmsAPI)
app.register(publicApiAuth)

// app.register(mediaPlugin)
// app.register(media)
// app.register(attachments, { uploadDir: "uploads" })

export default app
