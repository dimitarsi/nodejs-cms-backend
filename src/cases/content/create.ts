import { ObjectId } from "mongodb"
import type { ContentTypesRepo, ContentsRepo } from "~/repo"
import { generateFromConfig } from "./helpers"

export default function createContentCaseFrom(repo: {
  contents: ContentsRepo
  contentTypes: ContentTypesRepo
}) {
  return {
    async createContent(payload: Record<string, any>) {
      if (payload.data === null && payload.configId) {
        const config = await repo.contentTypes.getById(
          new ObjectId(payload.configId)
        )

        payload.data = config ? generateFromConfig(config) : null
      }

      const result = await repo.contents.create(payload)
      return await repo.contents.getById(result.insertedId.toString())
    },
  }
}
