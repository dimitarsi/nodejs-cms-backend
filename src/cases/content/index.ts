import { ObjectId } from "mongodb"
import type { ContentTypesRepo, ContentsRepo } from "~/repo"
import {
  emptyDataOrNull,
  generateFromConfig,
  validateContentWithConfig,
} from "./helpers"

const getDefaultPayload = (slug: string) => ({
  folderLocation: "/",
  folderTarget: slug,
})

export default function createContentCaseFrom(repo: {
  contents: ContentsRepo
  contentTypes: ContentTypesRepo
}) {
  return {
    async byId(
      idOrSlug: string,
      options: { validateData: boolean } = { validateData: true }
    ) {
      const entity = await repo.contents.getById(idOrSlug)

      // Nothing to validate, return the entity as is
      if (!options?.validateData) {
        return entity
      }

      // No config found!
      if (entity.configId === undefined) {
        return entity
      }

      const config = await repo.contentTypes.getById(entity.configId)

      // No config found!
      if (!config) {
        return entity
      }

      return validateContentWithConfig(entity, config)
    },
    async createContent(payload: Record<string, any>) {
      if (payload.isFolder) {
        const depth = Math.min(
          1,
          (payload.folderLocation || "/").split("/").length - 1
        )

        payload = {
          depth,
          ...getDefaultPayload(payload.slug),
          ...payload,
        }
      } else if (emptyDataOrNull(payload.data) && payload.configId) {
        const config = await repo.contentTypes.getById(
          new ObjectId(payload.configId)
        )

        payload.data = config ? generateFromConfig(config) : null
      }

      const result = await repo.contents.create(payload)
      return await repo.contents.getById(result.insertedId.toString())
    },
    async updateContent(idOrSlug: string, payload: Record<string, any>) {
      if (
        !payload.isFolder &&
        payload.configId &&
        emptyDataOrNull(payload.data)
      ) {
        const config = await repo.contentTypes.getById(
          new ObjectId(payload.configId)
        )

        payload.data = config ? generateFromConfig(config) : null
      }

      await repo.contents.update(idOrSlug, payload)

      return await repo.contents.getById(idOrSlug)
    },
  }
}
