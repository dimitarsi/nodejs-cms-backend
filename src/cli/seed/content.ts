import contentTypesRepo from "~/repo/contentTypes"
import { type Db, type ObjectId } from "mongodb"
import contentRepo from "~/repo/contents"
import slugify from "~/helpers/slugify"
import createContentCaseFrom from "~/cases/content"

export const createContent =
  (db: Db) =>
  async (
    configId: ObjectId | string | undefined,
    name?: string,
    data: any = null
  ) => {
    if (!configId || !name) {
      return
    }

    const contents = contentRepo(db)
    const contentTypes = contentTypesRepo(db)
    const usecase = createContentCaseFrom({ contents, contentTypes })

    return await usecase.createContent({
      name: name,
      slug: slugify(name),
      configId: configId,
      folderLocation: "/",
      isFolder: false,
      folderTarget: "/",
      children: [],
      data,
    })
  }
