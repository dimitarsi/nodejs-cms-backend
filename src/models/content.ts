import { ObjectId } from "mongodb";
import { BaseModel } from "./modal";
import { ContentType } from "./contentType";

type FileId = string;
type RefId = string;

export type TextDataType = {
  type: "text"
  value: string
}

export type FileDataType = {
  type: "files"
  value: FileId
}

export type RefDataType = {
  type: "ref"
  value: {
    refId: RefId
    type: string
  }
}

export type ComponentDataType = {
  type: "component"
  componentId: string | ObjectId
  value: Record<string, DataTypes>
}

export type DataTypes = TextDataType | FileDataType | RefDataType | ComponentDataType

export interface Content extends BaseModel {
  displayName: string
  id?: String
  configId: string | ObjectId
  config: ContentType
  data: Record<string, DataTypes>
}