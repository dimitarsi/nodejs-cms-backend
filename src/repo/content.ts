import { ObjectId } from "mongodb"
import makeRepo from "./crud"
import { Content } from "~/models/content"

const withUpdateConfigId = (data: Record<string, any>) => {
  const { configId = "", folderLocation, ...splat } = data
  const config = configId ? { configId: new ObjectId(configId) } : {}

  let folderConfig = {}

  if (folderLocation) {
    folderConfig = {
      folderLocation,
      depth: folderLocation.split("/").length - 1 || 1,
    }
  }

  return {
    ...folderConfig, // calculates the depth, in case we need it
    ...config,
    ...splat,
  }
}
/**
 * Returns the first item of an array,
 * needed because the aggregate
 * @param data
 * @returns
 */
const unwrapConfigField = (data: Content): Content => {
  const { config, ...splat } = data

  if (config && Array.isArray(config) && config.length >= 1) {
    return { config: config[0], ...splat }
  }

  return { config, ...splat }
}

const crud = makeRepo<Content>("content", { softDelete: false })
const collection = crud.getCollection()
const contentTypesDbName = crud.collectionName()

const Repo = {
  ...crud,
  create(data: Partial<Omit<Content, "id">>) {
    const normalized = withUpdateConfigId(data)

    return crud.create({
      createdOn: new Date(),
      updatedOn: new Date(),
      active: true,
      ...normalized,
    } as any)
  },
  update(id: string, data: Record<string, any>) {
    return crud.update(id, {
      updatedOn: new Date(),
      ...withUpdateConfigId(data),
    })
  },
  async getAll(
    page = 1,
    options: { pageSize: number; filter: Record<string, any> } = {
      pageSize: 20,
      filter: {},
    }
  ) {
    const filter = options.filter || {}

    const cursor = await collection.find(filter, {
      skip: options.pageSize * (page - 1),
      limit: options.pageSize,
    })

    const [items, count] = await Promise.all([
      cursor.toArray(),
      collection.countDocuments(filter),
    ])
    cursor.close()

    const totalPages = Math.ceil(count / options.pageSize)

    return {
      items,
      pagination: {
        page,
        perPage: options.pageSize,
        count,
        totalPage: totalPages,
        nextPage: page >= totalPages ? null : page + 1,
      },
    }
  },
  async getById(id: string) {
    let query: any = { _id: -1 }

    try {
      query = {
        $or: [{ _id: new ObjectId(id) }, { slug: id }],
      }
    } catch (_e) {
      query = { slug: id }
    }

    const cursor = await collection.aggregate([
      {
        $match: {
          ...query,
        },
      },
      {
        $lookup: {
          from: contentTypesDbName,
          localField: "configId",
          foreignField: "_id",
          as: "config",
        },
      },
    ])

    const data: any[] = await cursor.toArray()
    cursor.close()

    return data.map(unwrapConfigField)[0]
  },
}

export type ContentRepo = typeof Repo

export default Repo
