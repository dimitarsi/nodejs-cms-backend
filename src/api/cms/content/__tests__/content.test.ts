import { describe } from "@jest/globals"
import { MongoClient } from "mongodb"
import supertest from "supertest"
import app from "../../../../app"
import { CreateContentPayload } from "~/models/content"

const createContent = (): CreateContentPayload => ({
  name: "foo",
  slug: "foo",
  isFolder: false,
  folderLocation: "/",
  folderTarget: "",
  children: [],
  data: null,
})

describe("Content", () => {
  const mongoClient = new MongoClient(
    process.env.MONGO_URL || "mongodb://root:example@localhost:27017"
  )
  const db = mongoClient.db(process.env.DB_NAME)

  afterAll(() => {
    app.close()
    mongoClient.close()
  })

  afterEach(async () => {
    await db.collection("contents").deleteMany({})
  })

  describe("Needs Authentication", () => {
    test("GET /contents", async () => {
      await app.ready()

      await supertest(app.server).get("/contents").expect(403)
    })

    test("GET /contents/:id", async () => {
      await app.ready()

      await supertest(app.server).get("/contents/foobar").expect(403)
    })

    test("POST /contents", async () => {
      await app.ready()

      await supertest(app.server)
        .post("/contents")
        .send(createContent())
        .expect(403)
    })

    test("PATCH /contents/foobar", async () => {
      await app.ready()

      await supertest(app.server)
        .patch("/contents/foobar")
        .send({
          hello: "world",
        })
        .expect(403)
    })

    test("DELETE /contents/foobar", async () => {
      await app.ready()

      await supertest(app.server).delete("/contents/foobar").expect(403)
    })
  })

  describe("Only Admin users can create, update, delete and inspect content types", () => {
    describe("As Admin user", () => {
      test("GET /contents", async () => {
        await app.ready()

        supertest(app.server)
          .get("/contents")
          .set("X-Access-Token", process.env.TEST_ADMIN_ACCESS_TOKEN!)
          .expect(200)
      })

      test("POST /contents", async () => {
        await app.ready()

        const data = createContent()

        expect(data.isFolder).toBeDefined()

        const resp = await supertest(app.server)
          .post("/contents")
          .send(createContent())
          .set("X-Access-Token", process.env.TEST_ADMIN_ACCESS_TOKEN!)
          .expect(201)

        expect(resp.body).toBeDefined()
        expect(resp.body).toHaveProperty("_id")
        // expect(resp.body).toHaveProperty("type")
        expect(resp.body).toHaveProperty("name")
        expect(resp.body).toHaveProperty("slug")
        expect(resp.body).toHaveProperty("isFolder")
        expect(resp.body).toHaveProperty("folderTarget")
        expect(resp.body).toHaveProperty("folderLocation")
        expect(resp.body).toHaveProperty("folderDepth")
        expect(resp.body.folderDepth).toBe(0)
        // expect(resp.body).toHaveProperty("data")

        const { _id, createdOn, updatedOn, ...body } = resp.body
        expect(body).toMatchSnapshot()

        expect(resp.headers["location"]).toMatch(/^\/contents\/(.+)$/)
      })

      test("GET /contents/:idOrSlug", async () => {
        await app.ready()

        // Ensure entity exists
        const resp = await supertest(app.server)
          .post("/contents")
          .send(createContent())
          .set("X-Access-Token", process.env.TEST_ADMIN_ACCESS_TOKEN!)
          .expect(201)

        expect(resp.body).toBeDefined()

        const resultFromSlug = await supertest(app.server)
          .get(`/contents/${resp.body.slug}`)
          .set("X-Access-Token", process.env.TEST_ADMIN_ACCESS_TOKEN!)
          .expect(200)

        const resultFromId = await supertest(app.server)
          .get(`/contents/${resp.body._id}`)
          .set("X-Access-Token", process.env.TEST_ADMIN_ACCESS_TOKEN!)
          .expect(200)

        expect(resultFromId.body._id).toBeDefined()
        expect(resultFromSlug.body._id).toBeDefined()
        expect(resultFromId.body._id).toBe(resultFromSlug.body._id)
      })

      test("PATCH /contents/:idOrSlug", async () => {
        await app.ready()

        // Ensure entity exists
        const resp = await supertest(app.server)
          .post("/contents")
          .send(createContent())
          .set("X-Access-Token", process.env.TEST_ADMIN_ACCESS_TOKEN!)
          .expect(201)

        expect(resp.body).toBeDefined()

        await supertest(app.server)
          .patch(`/contents/${resp.body.slug}`)
          .set("X-Access-Token", process.env.TEST_ADMIN_ACCESS_TOKEN!)
          .send({
            name: "Hello World",
          })
          .expect(200)

        await supertest(app.server)
          .patch(`/contents/${resp.body._id}`)
          .send({
            name: "Hello World 2",
          })
          .set("X-Access-Token", process.env.TEST_ADMIN_ACCESS_TOKEN!)
          .expect(200)
      })

      test("PATCH /contents/:idOrSlug - changing foder also updates the depth", async () => {
        await app.ready()

        const data = createContent()

        // Ensure entity exists
        const resp = await supertest(app.server)
          .post("/contents")
          .send(data)
          .set("X-Access-Token", process.env.TEST_ADMIN_ACCESS_TOKEN!)
          .expect(201)

        expect(resp.body).toBeDefined()
        expect(resp.body.folderDepth).toBe(0)

        const response = await supertest(app.server)
          .patch(`/contents/${resp.body.slug}`)
          .set("X-Access-Token", process.env.TEST_ADMIN_ACCESS_TOKEN!)
          .send({
            name: "Hello World",
            folderLocation: "/posts",
          })
          .expect(200)

        expect(response.body.folderDepth).toBe(1)
        expect(response.body.name).toBe("Hello World")
        expect(response.body.slug).toBe(data.slug)
      })

      test("DELETE /contents/:id", async () => {
        await app.ready()

        // Ensure entity exists
        const resp = await supertest(app.server)
          .post("/contents")
          .send(createContent())
          .set("X-Access-Token", process.env.TEST_ADMIN_ACCESS_TOKEN!)
          .expect(201)

        expect(resp.body).toBeDefined()

        await supertest(app.server)
          .delete(`/contents/${resp.body._id}`)
          .set("X-Access-Token", process.env.TEST_ADMIN_ACCESS_TOKEN!)
          .expect(200)
      })

      test("DELETE /contents/:slug", async () => {
        await app.ready()

        // Ensure entity exists
        const resp = await supertest(app.server)
          .post("/contents")
          .send(createContent())
          .set("X-Access-Token", process.env.TEST_ADMIN_ACCESS_TOKEN!)
          .expect(201)

        expect(resp.body).toBeDefined()

        await supertest(app.server)
          .delete(`/contents/${resp.body.slug}`)
          .set("X-Access-Token", process.env.TEST_ADMIN_ACCESS_TOKEN!)
          .expect(200)
      })
    })

    describe("As non-Admin user", () => {
      test("GET /contents", async () => {
        await app.ready()

        supertest(app.server)
          .get("/contents")
          .set("X-Access-Token", process.env.TEST_NON_ADMIN_ACCESS_TOKEN!)
          .expect(403)
      })

      test("POST /contents", async () => {
        await app.ready()

        const resp = await supertest(app.server)
          .post("/contents")
          .send(createContent())
          .set("X-Access-Token", process.env.TEST_NON_ADMIN_ACCESS_TOKEN!)
          .expect(403)
      })

      test("GET /contents/:idOrSlug", async () => {
        await app.ready()

        // Ensure entity exists
        const resp = await supertest(app.server)
          .post("/contents")
          .send(createContent())
          .set("X-Access-Token", process.env.TEST_ADMIN_ACCESS_TOKEN!)
          .expect(201)

        expect(resp.body).toBeDefined()

        await supertest(app.server)
          .get(`/contents/${resp.body.slug}`)
          .set("X-Access-Token", process.env.TEST_NON_ADMIN_ACCESS_TOKEN!)
          .expect(403)

        await supertest(app.server)
          .get(`/contents/${resp.body._id}`)
          .set("X-Access-Token", process.env.TEST_NON_ADMIN_ACCESS_TOKEN!)
          .expect(403)
      })

      test("PATCH /contents/:idOrSlug", async () => {
        await app.ready()

        // Ensure entity exists
        const resp = await supertest(app.server)
          .post("/contents")
          .send(createContent())
          .set("X-Access-Token", process.env.TEST_ADMIN_ACCESS_TOKEN!)
          .expect(201)

        expect(resp.body).toBeDefined()

        await supertest(app.server)
          .patch(`/contents/${resp.body.slug}`)
          .set("X-Access-Token", process.env.TEST_NON_ADMIN_ACCESS_TOKEN!)
          .send({
            name: "Hello World",
          })
          .expect(403)

        await supertest(app.server)
          .patch(`/contents/${resp.body._id}`)
          .send({
            name: "Hello World 2",
          })
          .set("X-Access-Token", process.env.TEST_NON_ADMIN_ACCESS_TOKEN!)
          .expect(403)
      })

      test("DELETE /contents/:id", async () => {
        await app.ready()

        // Ensure entity exists
        const resp = await supertest(app.server)
          .post("/contents")
          .send(createContent())
          .set("X-Access-Token", process.env.TEST_ADMIN_ACCESS_TOKEN!)
          .expect(201)

        expect(resp.body).toBeDefined()

        await supertest(app.server)
          .delete(`/contents/${resp.body._id}`)
          .set("X-Access-Token", process.env.TEST_NON_ADMIN_ACCESS_TOKEN!)
          .expect(403)
      })

      test("DELETE /contents/:slug", async () => {
        await app.ready()

        // Ensure entity exists
        const resp = await supertest(app.server)
          .post("/contents")
          .send(createContent())
          .set("X-Access-Token", process.env.TEST_ADMIN_ACCESS_TOKEN!)
          .expect(201)

        expect(resp.body).toBeDefined()

        await supertest(app.server)
          .delete(`/contents/${resp.body.slug}`)
          .set("X-Access-Token", process.env.TEST_NON_ADMIN_ACCESS_TOKEN!)
          .expect(403)
      })
    })
  })

  describe("Quering content types", () => {
    test("find content by folder - `/contents?folder=/posts`", async () => {
      await app.ready()

      const data = createContent()
      const dataInFolder = createContent()

      await supertest(app.server)
        .post("/contents")
        .send(data)
        .set("X-Access-Token", process.env.TEST_ADMIN_ACCESS_TOKEN!)
        .expect(201)

      dataInFolder.slug = "test-folder-query"
      dataInFolder.folderLocation = "/posts"
      // dataInFolder.folderDepth = 2

      await supertest(app.server)
        .post("/contents")
        .send(dataInFolder)
        .set("X-Access-Token", process.env.TEST_ADMIN_ACCESS_TOKEN!)
        .expect(201)

      await supertest(app.server)
        .get(`/contents/${dataInFolder.slug}`)
        .set("X-Access-Token", process.env.TEST_ADMIN_ACCESS_TOKEN!)
        .expect(200)

      const response = await supertest(app.server)
        .get(`/contents?folder=/posts`)
        .set("X-Access-Token", process.env.TEST_ADMIN_ACCESS_TOKEN!)
        .expect(200)

      expect(response.body.items).toHaveLength(1)
      expect(response.body.pagination.count).toBe(1)
      expect(response.body.items[0].slug).toBe(dataInFolder.slug)
      expect(response.body.items[0].folderDepth).toBe(1)
    })
  })
})
