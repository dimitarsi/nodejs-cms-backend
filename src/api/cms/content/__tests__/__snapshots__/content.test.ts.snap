// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content Only Admin users can create, update, delete and inspect content types As Admin user POST /contents 1`] = `
{
  "active": true,
  "config": [],
  "data": null,
  "folderDepth": 0,
  "folderLocation": "/",
  "folderTarget": "",
  "isFolder": false,
  "name": "foo",
  "slug": "foo",
}
`;
